src/observe-tag.js

```diff
@@ -15,24 +33,32 @@ module.exports = tagFunction => {
 		let oldTag = tagResult
 		const removedElementWithFocusData = {
 			index: null,
+			tagName: null,
 			selectionStart: null,
 			selectionEnd: null,
-			selectionDirection: null
+			selectionDirection: null,
+			scrollLeft: null,
+			scrollTop: null
 		}
 
 		// remove oldTag first so that we unobserve before we re-observe
 		if (oldTag) {
 			// if there was focus, we need to figure out what element has it
-			const children = oldTag.querySelectorAll('*')
-			const parentAndChildrenNodes = [oldTag, ...children]
-			removedElementWithFocusData.index = parentAndChildrenNodes.findIndex(element => element === document.activeElement)
+			const allElements = parentAndChildrenElements(oldTag)
+			removedElementWithFocusData.index = allElements.findIndex(element => element === document.activeElement)
```

Comment:
Could we do a query selector here for active element, instead of checking against a large list created by a wildcard selector?

Reply:
The goal here is to find the index of the element, not the element itself (if that's all we needed, we could just use `document.activeElement`). I'm not sure that a more specific query selector would help... :thinking: thinking about it now, we could query selector on only things that have a positive tabIndex?

This might be just another one of those things where I want to do the performance test, and then fix as necessary.

Reply:
I do agree that if there are performance issues, it's that we have to query and iterate over all the children of an element before and after. I'm... not a fan either...